### PR TITLE
City search function

### PR DESCRIPTION
**Pull request by mrvanagas**
Added city search page and component, by entering part of city name, gets a drop down menu and shows a 7 day forecast 
image

To do - improve styling, added icon handling from API.

### CODE REVIEW

**Comment by mrvanagas** [comment]
Fixed destination branch

**Comment by JuliaShavyrova** [comment]
please resolve conflicts first

File: client/src/Components/MultiCard.styles.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+export const Container = styled.div`
+
+    display: flex;
+    flex-direction: row;
+    flex-wrap: wrap;
+    justify-content: space-between;
+    align-items: center;
+    align-content: center;
+    margin: 1rem 1rem 1rem 1rem;
```
Comment (JuliaShavyrova) [inline client/src/Components/MultiCard.styles.js]: style can be written shorter

File: client/src/Components/MultiCard.styles.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    padding: 1rem;
+`;
+
+export const CardContainer = styled.div`
+  background-color: gray;
+  border-radius: 0.5rem 0.5rem 0.5rem 0.5rem;
```
Comment (JuliaShavyrova) [inline client/src/Components/MultiCard.styles.js]: the same

**Comment by mrvanagas** [comment]
> please resolve conflicts first

Done, conflicts resolved.

File: client/src/Components/MultiCard.styles.js (hunk 1 above)
Comment (mrvanagas) [reply client/src/Components/MultiCard.styles.js]: Shortened style

File: client/src/Components/MultiCard.styles.js (hunk 2 above)
Comment (mrvanagas) [reply client/src/Components/MultiCard.styles.js]: Shortened style

File: client/src/Components/SearchBar.jsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  const [chooseCityForecast, setChooseCityForecast] = useState([]);
+  const [count, setCount] = useState(10)
+ 
+
+  useEffect(() => {
+    let timerID
+
+    if (inputValue.length > minCharacters) {
+      console.log('start of request')
+     timerID = setTimeout(async () =>{
```
Comment (JuliaShavyrova) [inline client/src/Components/SearchBar.jsx]: seems like your code needs auto-formatting (incorrect spacing, brackets, etc)

**Comment by mrvanagas** [comment]
Added debouncing and cleaned up code

File: client/src/API/api.js [hunk 4]
```diff
@@ -11,7 +11,7 @@ export async function getTokenFromAPI() {
   try {
     const response = await axios.get(TOKEN_URL);
     console.log(response);
```
Comment (JuliaShavyrova) [inline client/src/API/api.js]: please remove console logs if they are not important for user

File: client/src/Components/MultiCard.jsx [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import * as Style from './MultiCard.styles';
+import dateHelper from '../Utils/dateHelper';
+
+const MultiWeather = ({ data }) => {
+  return (
+    <Style.Container>
+      {data.map((info, index) => (
```
Comment (JuliaShavyrova) [inline client/src/Components/MultiCard.jsx]: seems like `info` can be destructured

File: client/src/Utils/dateHelper.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+function dateHelper(date) {
```
Comment (JuliaShavyrova) [inline client/src/Utils/dateHelper.js]: please rename function, it's not clear what exactly function helps with

File: client/src/Utils/dateHelper.js [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+    0: 'Monday',
+    1: 'Tuesday',
+    2: 'Wednesday',
+    3: 'Thursday',
+    4: 'Friday',
+    5: 'Saturday',
+    6: 'Sunday'
+  };
+
+  return weekDayMap[weekDay];
```
Comment (JuliaShavyrova) [inline client/src/Utils/dateHelper.js]: function can be simplified and implemented with no hardcoded array

File: client/src/Components/MultiCard.jsx (hunk 5 above)
Comment (mrvanagas) [reply client/src/Components/MultiCard.jsx]: Done

File: client/src/API/api.js (hunk 4 above)
Comment (mrvanagas) [reply client/src/API/api.js]: Removed logs in a few files, will do another sweep.

File: client/src/Utils/dateHelper.js (hunk 6 above)
Comment (mrvanagas) [reply client/src/Utils/dateHelper.js]: Renamed

File: client/src/Utils/dateHelper.js (hunk 7 above)
Comment (mrvanagas) [reply client/src/Utils/dateHelper.js]: Done, refactored.

File: client/src/Components/SearchBar.jsx (hunk 3 above)
Comment (mrvanagas) [reply client/src/Components/SearchBar.jsx]: Code should be formatted now

File: client/src/Components/SearchBar.jsx [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+    if (inputValue.length > minCharacters) {
+      timerID = setTimeout(async () => {
+        try {
+          const res = await weatherAPI.get(endpoints.GET_CITY(inputValue));
+          setResults(res.data.locations);
+        } catch (error) {
+          alert(error);
+        }
+      }
+        , 900)
```
Comment (JuliaShavyrova) [inline client/src/Components/SearchBar.jsx]: still see incorrect formatting, please use npm scripts and packages provided in your project folder

File: client/src/Utils/dateHelper.js [hunk 9]
```diff
@@ -0,0 +1,7 @@
+function dateToWeekdayHelper(date) {
+  const weekDay = new Date(date).toLocaleString('en-US', { weekday: 'long' });
+
+  return weekDay
```
Comment (JuliaShavyrova) [inline client/src/Utils/dateHelper.js]: no need to create variable just for returning it in next line

Comment (mrvanagas) [reply client/src/Utils/dateHelper.js]: Changed

File: client/src/Components/SearchBar.jsx (hunk 8 above)
Comment (mrvanagas) [reply client/src/Components/SearchBar.jsx]: Ran code formatter, should be now properly formatted

**Review by JuliaShavyrova — APPROVED** [review]

**Pull request merged**